.circleci/config.yml

```diff
@@ -0,0 +1,139 @@
+defaults: &defaults
+  working_directory: ~/loga
+  steps:
+    - checkout
+    - &restore_gems
+      restore_cache:
+        keys:
+          - v2-gems-{{ checksum "Gemfile" }}-{{ checksum "Appraisals"}}-{{ .Environment.CIRCLE_JOB }}
+          - v2-gems-{{ checksum "Gemfile" }}
+    - &bundle_install
+      run: bundle install --jobs 4 --path=vendor/bundle --retry 3
+    - run: bundle exec appraisal install
+    - &cache_gems
+      save_cache:
+        key: v2-gems-{{ checksum "Gemfile" }}-{{ checksum "Appraisals"}}-{{ .Environment.CIRCLE_JOB }}
+        paths:
+          - ~/loga/gemfiles/vendor/bundle
+          - ~/loga/vendor/bundle
+    - attach_workspace:
+        at: ./tmp
+    - run: RACK_ENV=development bundle exec appraisal rspec
+    - run: RACK_ENV=production bundle exec appraisal rspec
+    - run: ./tmp/cc-test-reporter format-coverage -t simplecov -o $(ruby -e 'puts "tmp/codeclimate.#{RUBY_VERSION}.json"') coverage/.resultset.json
+    - persist_to_workspace:
+        root: tmp
+        paths:
+          - codeclimate.*.json
+    - store_artifacts:
+        path: coverage
+
+version: 2
+jobs:
+  build:
+    working_directory: ~/loga
+    docker:
+      - image: circleci/ruby:2.5.1
+    steps:
+      - run:
+          name: Download cc-test-reporter
+          command: |
+            mkdir -p tmp/
```

Comment:
🙋 If the sole purpose of creating `tmp/` is to store the `cc-test-reporter` binary then it's not necessary. You can cache the file using `root: .`.

Reply:
I would like not to pollute the working directory with some random scripts. There isn't a `checkout` here but we could add one in the future 🤷‍♂️

Also everywhere it is the workspace is attached at `tmp` and it feels less magical to me with this line :)